tests/test_plugin_loader.py

```
@@ -4463,5 +4738,163 @@ def test_update_existing_repo_to_branch_or_tag_fetch_failure(
+
+        src = inspect.getsource(pl)
+        self.assertIn(
+            "_SYS_MODULES_LOCK", src, "Lock name should appear in module source"
+        )
+
+    def test_exec_plugin_module_raises_import_error_when_no_loader(self):
+        import importlib.machinery
+
+        spec = importlib.machinery.ModuleSpec("_test_no_loader", loader=None)
+        module = ModuleType("_test_no_loader")
+        with self.assertRaises(ImportError) as ctx:
```

Ruff (0.15.7)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)